### PR TITLE
[CVS.com] New ruleset, #5599

### PR DESCRIPTION
**Pull request by chris-barry**

### CODE REVIEW

**Comment by chris-barry** [comment]
I have a feeling I need to add a test url. I will look in to it tonight.

**Comment by fuglede** [comment]
It's not the lack of tests that's the issue here, but the fact that https://www.cvs.com/ serves a 403.

    ERROR src/chrome/content/rules/CVS.com.xml: Non-2xx HTTP code: http://cvs.com/ (200) => https://cvs.com/ (403)

**Comment by chris-barry** [comment]
When wgetting and watching the dev console, cvs.com is 302ing to www.cvs.com. I am not sure why a 403 occurred in the Travis test.

However, when I load the page with the most recent version of the Tor Browser Bundle, I see a 403. Maybe CVS is just blocking tor users based on the user agent?

**Comment by fuglede** [comment]
I haven't been able to not get a 403 in any of the browsers I've tried.

**Comment by chris-barry** [comment]
Could it be based on where they think the IP is geographically? I am trying
from the United States and see a 302 from cvs.com->www.cvs.com

**Comment by chris-barry** [comment]
Just took this screenshot about a minute ago.

cvs

**Comment by chris-barry** [comment]
And another with wget: 
cvs-cli

**Comment by fuglede** [comment]
That's certainly an explanation; with Tor, I'm seeing the following

    $ curl -I https://www.cvs.com
    HTTP/1.1 403 Forbidden
    $ curl --socks5 localhost:9050 -I https://www.cvs.com
    HTTP/1.1 200 OK
    // Renew Tor route
    $ curl --socks5 localhost:9050 -I https://www.cvs.com
    HTTP/1.1 403 Forbidden

The machines I used earlier were placed at various places in Europe. Kind of crazy if they blocked out all non-US countries though.

**Comment by chris-barry** [comment]
ayyy cvs.com seems to work with https now :D. Any chance of a merge?

**Comment by gloomy-ghost** [comment]
Travis is passing but I can't confirm it since cvs.com is blocking all non-US countries. 
@jeremyn @J0WI: Can you have a look at this? Also please close #5599 after this gets merged.

File: src/chrome/content/rules/CVS.com.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	<target host="m.cvs.com" />
+
+	<target host="cvslearnet.cvs.com" />
+	<target host="es.cvs.com" />
+	<target host="federation.cvs.com" />
+	<target host="vaccines.cvs.com" />
+	<target host="vendorportal.cvs.com" />
+
+	<rule from="^http:" to="https:" />
+</ruleset>
```
Comment (J0WI) [inline src/chrome/content/rules/CVS.com.xml]: See also:
```
flushot.cvs.com
link.cvs.com

File: src/chrome/content/rules/CVS.com.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!--
+	Mixed:
+	- health
+	- optical
```
Comment (J0WI) [inline src/chrome/content/rules/CVS.com.xml]: I didn't found major MCB issues. Do you have any examples?

**Comment by jeremyn** [comment]
@chris-barry Are you still interested in working on this?

**Comment by chris-barry** [comment]
@jeremyn: yes. I haven't touched it due to CVS blocking non-US countries with https. If you feel this rule will be included I will work on it

**Comment by jeremyn** [comment]
I'd like to have this resolved and can take a look at your updates. However if HTTPS fails outside of the US for some reason then we will need to make the ruleset `default_off` (disabled by default).

**Comment by J0WI** [comment]
> I haven't touched it due to CVS blocking non-US countries with https.

AFAIKS this is not limited to httpS, so the behavior will be consistent.

**Comment by chris-barry** [comment]
I have rebased and modified my PR to take in to account feedback.

**Comment by jeremyn** [comment]
@chris-barry Please make the following changes:

Delete entirely:
- [ ] https://federation.cvs.com (`refused` with `https`, `time out` with `http`)

Mark as `Different HTTP/HTTPS content`:
- [ ] https://photo-metrics.cvs.com
- [ ] https://photo-smetrics.cvs.com

Mark as `Invalid certificate`:
- [ ] https://link.cvs.com (and remove from `Mixed`)
- [ ] https://s.cvs.com

Mark as `No working URL known`:
- [ ] https://depservices.cvs.com
- [ ] https://photohelp.cvs.com
- [ ] https://services.cvs.com
- [ ] https://static.cvs.com
- [ ] https://tnl-photo.cvs.com

Mark as `Redirects to HTTP`:
- [ ] https://health.cvs.com (and remove from `Mixed`)

Add `target`:
- [ ] https://circular.cvs.com with `test` http://circular.cvs.com/images/hosted/v4/carousel_up_arrow.png
- [ ] https://d.cvs.com
- [ ] https://ddl.cvs.com with `test` http://ddl.cvs.com/robots.txt
- [ ] https://express.cvs.com
- [ ] https://i.cvs.com
- [ ] https://message.cvs.com
- [ ] https://myhr.cvs.com
- [ ] https://mypassword.cvs.com
- [ ] https://photo-store.cvs.com
- [ ] https://t.cvs.com

Also:
- [ ] Sort the comments and `target`s as described in [`CONTRIBUTING.md#target-ordering`](https://github.com/EFForg/https-everywhere/blob/45cf7c85151609862175b09c8ee682d204846465/CONTRIBUTING.md#target-ordering)

Tests should look like this:
```xml
	<target host="circular.cvs.com" />
		<test url="http://circular.cvs.com/images/hosted/v4/carousel_up_arrow.png">
	<target host="..." />
```

Items in the top comment should look like this:
```xml
<!--
	Different HTTP/HTTPS content:
		photo-metrics.cvs.com
		photo-smetrics.cvs.com

	Invalid certificate:
		...

	...

-->
```

Please let me know if you have any questions.

**Comment by jeremyn** [comment]
@chris-barry I'd like to keep this pull request moving along, do you think you'll be able to look at it soon?

**Comment by jeremyn** [comment]
@chris-barry I want to finish this pull request. If we don't hear from you soon, I'm going to go ahead and make a new pull request myself for these subdomains, and close this one.

**Comment by jeremyn** [comment]
I'm closing this pull request in favor of my new pull request https://github.com/EFForg/https-everywhere/pull/14552.

**Pull request closed without merging**